daemon/firewall/firewall_manager.go

```diff
@@ -0,0 +1,599 @@
+	}
+
+	// -I INPUT -i <iface> -m comment --comment nordvpn -j DROP
+	// -I OUTPUT -o <iface> -m comment --comment nordvpn -j DROP
+	for _, iface := range interfaces {
+		inputCommand := fmt.Sprintf("-I INPUT -i %s -m comment --comment nordvpn -j DROP", iface.Name)
+		if err := f.commandExecutor.ExecuteCommand(inputCommand); err != nil {
+			return fmt.Errorf("blocking input traffic: %w", err)
+		}
+
+		outputCommand := fmt.Sprintf("-I OUTPUT -o %s -m comment --comment nordvpn -j DROP", iface.Name)
+		if err := f.commandExecutor.ExecuteCommand(outputCommand); err != nil {
+			return fmt.Errorf("blocking output traffic: %w", err)
+		}
+	}
+
+	return nil
+}
+
+func (f *FirewallManager) BlockTraffic() error {
+	if f.trafficBlocked {
+		return fmt.Errorf("traffic is already blocked")
+	}
+
+	if f.enabled {
+		if err := f.addBlockTrafficRules(); err != nil {
+			return fmt.Errorf("adding block traffic rules: %w", err)
+		}
+	}
+
+	f.trafficBlocked = true
+
+	return nil
+}
+
+func (f *FirewallManager) removeBlockTrafficRules() error {
+	interfaces, err := f.devices()
+	if err != nil {
+		return fmt.Errorf("listing interfaces: %w", err)
+	}
+
+	// -D INPUT -i <iface> -m comment --comment nordvpn -j DROP
+	// -D OUTPUT -o <iface> -m comment --comment nordvpn -j DROP
+	for _, iface := range interfaces {
+		inputCommand := fmt.Sprintf("-D INPUT -i %s -m comment --comment nordvpn -j DROP", iface.Name)
+		if err := f.commandExecutor.ExecuteCommand(inputCommand); err != nil {
+			return fmt.Errorf("unblocking input traffic: %w", err)
+		}
+
+		outputCommand := fmt.Sprintf("-D OUTPUT -o %s -m comment --comment nordvpn -j DROP", iface.Name)
+		if err := f.commandExecutor.ExecuteCommand(outputCommand); err != nil {
+			return fmt.Errorf("unblocking output traffic: %w", err)
+		}
+	}
+
+	return nil
+}
+
+func (f *FirewallManager) UnblockTraffic() error {
+	if !f.trafficBlocked {
+		return fmt.Errorf("traffic is already unblocked")
+	}
+
+	if f.enabled {
+		if err := f.removeBlockTrafficRules(); err != nil {
+			return fmt.Errorf("removing traffic block rules: %w", err)
+		}
+	}
+
+	f.trafficBlocked = false
+
+	return nil
+}
+
+func (f *FirewallManager) addIncomingRule(rule allowIncomingRule) error {
+	if err := f.commandExecutor.ExecuteCommand("-I " + rule.allowIncomingRule); err != nil {
+		return fmt.Errorf("adding allow incoming rule: %w", err)
+	}
+
+	for _, blockLANRule := range rule.blockLANRules {
+		if err := f.commandExecutor.ExecuteCommand("-I " + blockLANRule); err != nil {
+			return fmt.Errorf("adding block peer lan rule: %w", err)
+		}
+	}
+
+	return nil
+}
+
+func (f *FirewallManager) AllowIncoming(peer meshnet.UniqueAddress, allowLocal bool) error {
+	if _, ok := f.allowIncomingRules[peer.UID]; ok {
+		return fmt.Errorf("allow incoming rule for the given peer already exists")
+	}
+
+	rule := fmt.Sprintf("INPUT -s %s/32 -m comment --comment nordvpn -j ACCEPT", peer.Address)
+
+	blockLANRules := []string{}
+	if !allowLocal {
+		lans := []string{
```

Comment:
What would happen if the user doesn't have one of the RFC IP ranges on the LAN?

Reply:
LAN block will not work for them, i.e it will be routable. I guess we could(or even should) make this more sophisticated with arp, but this it out of scope for this PR.